Sources/SmokeHTTP1/HTTP1ChannelInboundHandler.swift

```diff
@@ -29,51 +29,183 @@ class HTTP1ChannelInboundHandler: ChannelInboundHandler {
     typealias InboundIn = HTTPServerRequestPart
     typealias OutboundOut = HTTPServerResponsePart
     
+    private struct WaitingForRequestBody {
+        let logger: Logger
+        let internalRequestId: String
+        let requestHead: HTTPRequestHead
+        let keepAliveStatus: KeepAliveStatus
+        
+        init(requestHead: HTTPRequestHead) {
+            self.internalRequestId = UUID().uuidString
+            var newLogger = Logger(label: "com.amazon.SmokeFramework.request.\(internalRequestId)")
+            newLogger[metadataKey: "internalRequestId"] = "\(internalRequestId)"
+            
+            self.logger = newLogger
+            self.requestHead = requestHead
+            self.keepAliveStatus = KeepAliveStatus(state: requestHead.isKeepAlive)
+        }
+    }
+    
+    private struct ReceivingRequestBody {
+        let logger: Logger
+        let internalRequestId: String
+        let requestHead: HTTPRequestHead
+        let keepAliveStatus: KeepAliveStatus
+        let partialBody: Data
+        
+        init(waitingForRequestBody: WaitingForRequestBody, bodyPart: Data) {
+            self.logger = waitingForRequestBody.logger
+            self.internalRequestId = waitingForRequestBody.internalRequestId
+            self.requestHead = waitingForRequestBody.requestHead
+            self.keepAliveStatus = waitingForRequestBody.keepAliveStatus
+            self.partialBody = bodyPart
+        }
+        
+        init(receivingRequestBody: ReceivingRequestBody, bodyPart: Data) {
+            self.logger = receivingRequestBody.logger
+            self.internalRequestId = receivingRequestBody.internalRequestId
+            self.requestHead = receivingRequestBody.requestHead
+            self.keepAliveStatus = receivingRequestBody.keepAliveStatus
+            
+            var newPartialBody = receivingRequestBody.partialBody
+            newPartialBody += bodyPart
+            self.partialBody = newPartialBody
```

Comment:
Would it be possible to turn this into the following?
`self.partialBody = receivingRequestBody.partialBody + bodyPart`

Reply:
That should work. Thanks! I've updated the PR.